app/controllers/searches_controller.rb

```diff
@@ -383,16 +383,29 @@ def build_tweet_items(items)
     Kaminari.paginate_array(items.map { |t| {target: t} }).page(params[:page]).per(100)
   end
 
+  def fingerprint
+    if session[:fingerprint].nil? || session[:fingerprint] == -1
+      session[:fingerprint] = (session.id.nil? ? -1 : session.id)
+    else
+      session[:fingerprint]
+    end
+  end
+
   def create_log
-    SearchLog.create(
-      session_id: session.id,
-      user_id: current_user_id,
-      uid: @searched_tw_user.uid,
-      screen_name: @searched_tw_user.screen_name,
-      action: action_name,
-      ego_surfing: (user_signed_in? && current_user.uid.to_i == @searched_tw_user.uid.to_i))
+    SearchLog.create!(
```

Comment:
__[rubocop]__
* Assignment Branch Condition size for create_log is too high. [34.54/15] [:ref](http://c2.com/cgi/wiki?AbcMetric)
* Cyclomatic complexity for create_log is too high. [7/6]
* Method has too many lines. [14/10] [:ref](https://github.com/bbatsov/ruby-style-guide#short-methods)